x/ibc/02-client/handler.go

```diff
@@ -31,11 +31,7 @@ func HandleMsgCreateClient(ctx sdk.Context, k Keeper, msg exported.MsgCreateClie
 		}
 	case exported.Localhost:
 		// msg client id is always "localhost"
-		clientState = localhosttypes.NewClientState(
-			k.ClientStore(ctx, msg.GetClientID()),
-			ctx.ChainID(),
-			ctx.BlockHeight(),
-		)
+		clientState = localhosttypes.NewClientState(ctx.ChainID(), ctx.BlockHeight())
```

Comment:
this looks a lot cleaner :raised_hands: I think it'll be pretty straight forward to create interfaces funcs for clients to implement so we can remove custom client handling here (see #6064)